mock_tests/test_auth.py

```diff
@@ -140,17 +140,16 @@ def test_auth_header_without_weaviate_auth(weaviate_mock):
 
 def test_auth_header_with_catchall_proxy(weaviate_mock, recwarn):
     """Test that the client can handle situations in which a proxy returns a catchall page for all requests."""
-    bearer_token = "OTHER TOKEN"
-    weaviate_mock.expect_request(
-        "/v1/schema", headers={"Authorization": "Bearer " + bearer_token}
-    ).respond_with_json({})
+    weaviate_mock.expect_request("/v1/schema").respond_with_json({})
     weaviate_mock.expect_request("/v1/.well-known/openid-configuration").respond_with_data(
         "JsonCannotParseThis"
     )
 
     client = weaviate.Client(
         url=MOCK_SERVER_URL,
-        additional_headers={"Authorization": "Bearer " + bearer_token},
+        auth_client_secret=weaviate.AuthClientPassword(
+            username="test-username", password="test-password"
+        ),
```

Comment:
What is the reason for this change? The test was especially for setups without OIDC where a catchall proxy would return a login page (eg not valid json) for all urls

Reply:
the error that is expected in this test is thrown [here](https://github.com/weaviate/weaviate-python-client/blob/main/weaviate/connect/connection.py#L149), right? 

when the auth header is now passed via `additional_headers`, the `create_session` method returns directly. The OIDC config endpoint is not called (as the `_create_session` directly returns) and this error is never thrown. The method now perform an `get OIDC config` request only when the `auth_client_secret` is unequal to `AuthAPIKey` or  when the `Authorization` header is not present in the `self._headers` dict. 

So I just used a different Auth method so that the `get OIDC config` request is performed and the expected error is thrown.

Is this correct or did I misunderstood something?

Reply:
ahhh ok.

So the test was mainly to make sure that a custom authorization header would work behind a catch-all proxy. With your change it will definitely do.

I'm wondering if it makes sense to continue when the client was explicitly configured to do OIDC and weaviate then returns an invalid config 🤔  But that would be out of scope for this PR